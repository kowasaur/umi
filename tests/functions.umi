String createHello(String name) {
    # Implicit return
    "Hello, " + name
}

Void greet(String name, Int age) {
    String message = createHello(name)
    consoleWrite(message)
    message = " aged "
    consoleWrite(message)
    print(age)
}

# Example of function overloading
Void greet(Int age) {
    greet("Tim", age)
}

Void main() {
    greet("Joe", 3333)
    greet(52)
}
