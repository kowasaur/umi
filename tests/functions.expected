Hello, Joe aged 3333
Hello, Tim aged 52
You suck Bob
